src/Chapter3.hs

```diff
@@ -480,6 +519,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+
+data Building = Church | Library deriving (Show)
+data House = One | Two | Three | Four deriving (Enum, Show)
+
+data City = City
+    { cityCastle :: Maybe Castle
+    , cityWalls  :: Bool
```

Comment:
This is the correct data type, that can represent what we have in requirements 👍🏼 
However, this also allows having some states that should not ideally exist: like having walls without any castle. With this representation, this is possible. However, we actually can restrict this on type-level by providing more representative types 🙂 This is actually the greatest power of Haskell and type system 😍